src/frontend/src/optimizer/property/monotonicity.rs

```diff
@@ -186,6 +186,23 @@ impl MonotonicityAnalyzer {
             tz_is_utc // conservative
         }
 
+        fn date_unit_larger_than_day(date_unit: Option<&str>) -> bool {
+            const DAY: &str = "day";
+            const WEEK: &str = "week";
+            const MONTH: &str = "month";
+            const QUARTER: &str = "quarter";
+            const YEAR: &str = "year";
+            const DECADE: &str = "decade";
+            const CENTURY: &str = "century";
+            const MILLENNIUM: &str = "millennium";
+            date_unit.map_or(false, |date_unit| {
+                matches!(
+                    date_unit.to_ascii_lowercase().as_str(),
+                    DAY | WEEK | MONTH | QUARTER | YEAR | DECADE | CENTURY | MILLENNIUM
+                )
+            })
```

Comment:
what's the point of the `const` here

Reply:
In case SQL is standardized to use a language other than English in the future?

Reply:
just maintain it same as the expression's implementation https://github.com/risingwavelabs/risingwave/blob/c06b0c6ba7009855e2a43630a8e3ceb4a5ead6ba/src/expr/impl/src/scalar/date_trunc.rs#L20-L33

Reply:
This modification is a bit troublesome; we have to expose these const from a very deep level within expr... Since we might never modify it, I would rather make a copy.

Reply:
I am too lazy....

Reply:
Your brain will become rusty!